src/cascadia/TerminalCore/TerminalApi.cpp

```diff
@@ -126,6 +126,56 @@ COORD Terminal::GetCursorPosition()
     return newPos;
 }
 
+// Method Description:
+// - deletes uiCount characters starting from the cursor's current position
+// - it moves over the remaining text to 'replace' the deleted text
+// - for example, if the buffer looks like this ('|' is the cursor): [abc|def]
+// - calling DeleteCharacter(1) will change it to: [abc|ef],
+// - i.e. the 'd' gets deleted and the 'ef' gets shifted over 1 space and **retain their previous text attributes**
+// Arguments:
+// - uiCount, the number of characters to delete
+// Return value:
+// - true if succeeded, false otherwise
+bool Terminal::DeleteCharacter(const unsigned int uiCount)
+{
+    SHORT dist;
+    if (!SUCCEEDED(UIntToShort(uiCount, &dist)))
+    {
+        return false;
+    }
+    const auto cursorPos = _buffer->GetCursor().GetPosition();
+    const auto copyToPos = cursorPos;
+    COORD copyFromPos{ cursorPos.X + dist, cursorPos.Y };
+    auto sourceWidth = _mutableViewport.RightExclusive() - copyFromPos.X;
+    SHORT width;
+    if (!SUCCEEDED(UIntToShort(sourceWidth, &width)))
+    {
+        return false;
+    }
+
+    // Get a rectangle of the source
+    auto source = Viewport::FromDimensions(copyFromPos, width, (SHORT)1);
+    const auto sourceOrigin = source.Origin();
+
+    // Get a rectangle of the target
+    const auto target = Viewport::FromDimensions(copyToPos, source.Dimensions());

```

Comment:
Did you get a sample of this code from somewhere else? Would it make sense to consolidate this usage and the other one into a method on `TextBuffer` itself called `MoveRegion` or something?

Reply:
This would definitely be a good idea, will make it  a work item for now so we can get this in for 0.3 tomorrow